doc/development/deprecations.rst

```diff
@@ -6,6 +6,20 @@ Deprecations
 Pending deprecations
 --------------------
 
+* ``single_tape_transform``, ``batch_transform``, ``qfunc_transform``, and ``op_transform`` are
+  deprecated. Instead switch to using the new ``qml.transform`` function. Please refer to
+  `the transform docs <https://docs.pennylane.ai/en/stable/code/qml_transforms.html#custom-transforms>`_
+  to see how this can be done.
+
+  - Deprecated in v0.34
+  - Will be removed in v0.35
```

Comment:
Given the scope of these objects, I think we should probably have a longer deprecation cycle to allow people to more smoothly port.  0.36 maybe?